### ISSUE TITLE
config({timeout:5000}) not work

### ISSUE THREAD

**Issue by denghejun** [wkh237/react-native-fetch-blob#368]
Hi I found the timeout configuration not work for me:

config({timeout:5000}).fetch('GET','some_uri')

**Comment by dmdevoss** [comment]
On iOS:
I know I'm late to the party, but the issue is still open so I'll give my solution. 
TLDR: The **timeout** configuration is only honored if you also assign **followRedirect** to false.

The best I can tell, the NSURLSessionConfiguration **backgroundSessionConfiguration** doesn't even look at **timeoutIntervalForRequest** which is what is being set by this package when you edit the config object. It's suppose to instead look at **timeoutIntervalForResource**, however, [the default for that is 7 days according to apple](https://developer.apple.com/documentation/foundation/nsurlsessionconfiguration/1408153-timeoutintervalforresource?language=objc), which is either false, or my default is being overwritten by another source (not this package). I have tried manually setting it, but it didn't take. I'm a little out of my element looking at objective-c so I'm going to just stick with my solution for now. By setting  **followRedirect** to false, this package uses the **defaultSessionConfiguration** rather than the background one. Obviously this solution wont work for everyone since there are other things that setting **followRedirect** to false does.

**Comment by wkh237** [comment]
@dmdevoss , thank you for the great explanation 👍 Will try to fix this bug, and of course, a PR is always welcomed !

**Comment by binbinla** [comment]
hello，I found that `timeout` did not work on android too .  Is that an existing problem?

**Comment by binbinla** [comment]
Here is what I set : `config({timeout:15000})`,  but when I tested on the device,  the timeout notification appeared after 30000, it's double than what I  actually want.
